dotnet/src/Connectors/Connectors.Memory.CosmosNoSql/CosmosNoSqlServiceCollectionExtensions.cs

```diff
@@ -63,24 +63,26 @@ public static IServiceCollection AddCosmosNoSqlVectorStore(
         CosmosNoSqlVectorStoreOptions? options = default,
         string? serviceId = default)
     {
+        Verify.NotNullOrWhiteSpace(connectionString);
```

Comment:
nit: don't we already verify these inside the constructor for CosmosNoSqlVectorStore below? If we don't, we should... The DI registration methods shouldn't in principle need to do validation - that should be the responsibility of the types the instantiate etc.

Reply:
> don't we already verify these inside the constructor for CosmosNoSqlVectorStore below?

We do, but we register a factory method that creates `CosmosNoSqlVectorStore` and this factory method is being called when the type is requested for the first time. So in case the user has provided null connection string, we won't throw immediately, but later.

I personally believe we should perform simple argument validation and reject invalid inputs immediately rather than waiting for the object to be requested.

Reply:
OK. Yeah, the fact that in this DI context the validation is deferred is a good argument.